### PR TITLE
updated upstream version to v3.10.11

### PR DESCRIPTION
**Pull request by joelkraehemann**
* implemented VST3 support

### CODE REVIEW

**Comment by hfiguiere** [comment]
You should add the content of #120

File: gtkmm.yaml [hunk 1]
```diff
@@ -0,0 +1,87 @@
+name: gtkmm
+sources:
+  - sha256: 6d71091bcd1863133460d4188d04102810e9123de19706fb656b7bb915b4adc3
+    type: archive
+    url: http://ftp.gnome.org/pub/GNOME/sources/gtkmm/3.24/gtkmm-3.24.2.tar.xz
```
Comment (hfiguiere) [inline gtkmm.yaml:5]:
```suggestion
    url: https://ftp.gnome.org/pub/GNOME/sources/gtkmm/3.24/gtkmm-3.24.2.tar.xz
```

File: gtkmm.yaml [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+  - /share/aclocal
+  - /share/man
+  - /share/pkgconfig
+
+modules:
+  - name: sigc++
+    sources:
+      - sha256: 0b68dfc6313c6cc90ac989c6d722a1bf0585ad13846e79746aa87cb265904786
+        type: archive
+        url: http://ftp.gnome.org/pub/GNOME/sources/libsigc++/2.10/libsigc++-2.10.3.tar.xz
```
Comment (hfiguiere) [inline gtkmm.yaml:25]:
```suggestion
        url: https://ftp.gnome.org/pub/GNOME/sources/libsigc++/2.10/libsigc++-2.10.3.tar.xz
```

File: gtkmm.yaml [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+      - '*.la'
+      - /include/sigc++-2.0
+      - /lib/pkgconfig
+      - /lib/sigc++-2.0
+
+  - name: glibmm
+    sources:
+      - sha256: a75282e58d556d9b2bb44262b6f5fb76c824ac46a25a06f527108bec86b8d4ec
+        type: archive
+        url: http://ftp.gnome.org/pub/GNOME/sources/glibmm/2.64/glibmm-2.64.2.tar.xz
```
Comment (hfiguiere) [inline gtkmm.yaml:38]:
```suggestion
        url: https://ftp.gnome.org/pub/GNOME/sources/glibmm/2.64/glibmm-2.64.2.tar.xz
```

File: gtkmm.yaml [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+      - /lib/glibmm-2.4
+      - /lib/giomm-2.4
+      - /lib/libglibmm_generate_extra_defs*
+      - /lib/pkgconfig
+
+  - name: cairomm
+    sources:
+      - sha256: a54ada8394a86182525c0762e6f50db6b9212a2109280d13ec6a0b29bfd1afe6
+        type: archive
+        url: http://ftp.gnome.org/pub/GNOME/sources/cairomm/1.12/cairomm-1.12.0.tar.xz
```
Comment (hfiguiere) [inline gtkmm.yaml:54]:
```suggestion
        url: https://ftp.gnome.org/pub/GNOME/sources/cairomm/1.12/cairomm-1.12.0.tar.xz
```

File: gtkmm.yaml [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+      - '*.la'
+      - /include/cairomm-1.0
+      - /lib/cairomm-1.0
+      - /lib/pkgconfig
+
+  - name: pangomm
+    sources:
+      - sha256: 14bf04939930870d5cfa96860ed953ad2ce07c3fd8713add4a1bfe585589f40f
+        type: archive
+        url: http://ftp.gnome.org/pub/GNOME/sources/pangomm/2.42/pangomm-2.42.1.tar.xz
```
Comment (hfiguiere) [inline gtkmm.yaml:67]:
```suggestion
        url: https://ftp.gnome.org/pub/GNOME/sources/pangomm/2.42/pangomm-2.42.1.tar.xz
```

File: gtkmm.yaml [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+      - '*.la'
+      - /include/pangomm-1.4
+      - /lib/pangomm-1.4
+      - /lib/pkgconfig
+
+  - name: atkmm
+    sources:
+    - sha256: 4c4cfc917fd42d3879ce997b463428d6982affa0fb660cafcc0bc2d9afcedd3a
+      type: archive
+      url: http://ftp.gnome.org/pub/GNOME/sources/atkmm/2.28/atkmm-2.28.0.tar.xz
```
Comment (hfiguiere) [inline gtkmm.yaml:80]:
```suggestion
      url: https://ftp.gnome.org/pub/GNOME/sources/atkmm/2.28/atkmm-2.28.0.tar.xz
```

File: org.nongnu.gsequencer.gsequencer.json [hunk 7]
```diff
@@ -154,11 +156,41 @@
...
+            "buildsystem": "cmake-ninja",
+            "builddir": true,
+            "build-options": {
+                "env": {
+                    "CXXFLAGS": "-include cstdio"
+                }
+            },
+            "build-commands": [
+                "cmake -DCMAKE_BUILD_TYPE=RelWithDebugInfo ..",
+                "ninja",
```
Comment (hfiguiere) [inline org.nongnu.gsequencer.gsequencer.json:170]:
```suggestion
            "post-install": [
```

Comment (hfiguiere) [reply org.nongnu.gsequencer.gsequencer.json:170]: you have buildsystem = ninja so build-commands is ignored. (there should even be a warning)

**Review by hfiguiere — CHANGES_REQUESTED** [review]

**Comment by hfiguiere** [comment]
`cannot find -lstdc++fs` <- that's the build system that assume they run on Ubuntu.

**Comment by joelkraehemann** [comment]
> `cannot find -lstdc++fs` <- that's the build system that assume they run on Ubuntu.

howto fix it?

File: org.nongnu.gsequencer.gsequencer.json [hunk 8]
```diff
@@ -154,11 +154,42 @@
...
+            "name": "vst3sdk",
+            "buildsystem": "cmake-ninja",
+            "builddir": true,
+            "config-opts": [
+                "-DCMAKE_BUILD_TYPE=RelWithDebugInfo"
+            ],
+            "build-options": {
+                "env": {
+                    "CXXFLAGS": "-include cstdio"
+                }
```
[the remark this answers is not in the document]
Comment (joelkraehemann) [reply org.nongnu.gsequencer.gsequencer.json]: this doesn't work how you expected it, sorry.

**Comment by joelkraehemann** [comment]
bot, build

**Comment by hfiguiere** [comment]
I have a working local build but it just terminate without message after loading sfizz.

```
** Message: 06:32:31.017: ags_vst3_manager.c loading - /app/extensions/Plugins/vst3/sfizz.vst3/Contents/x86_64-linux/sfizz.so
[sfizz] new synth
```

(it's `org.freedesktop.LinuxAudio.Plugins.sfizz`)

**Comment by hfiguiere** [comment]
actually it's a SEGV

```
Thread 4 "gsequencer" received signal SIGSEGV, Segmentation fault.
[Switching to Thread 0x7fffe77fe700 (LWP 14)]
0x00007fffcc20d5a5 in ?? () from /app/extensions/Plugins/vst3/sfizz.vst3/Contents/x86_64-linux/sfizz.so
(gdb) where
#0  0x00007fffcc20d5a5 in  () at /app/extensions/Plugins/vst3/sfizz.vst3/Contents/x86_64-linux/sfizz.so
#1  0x00007fffcc20f077 in  () at /app/extensions/Plugins/vst3/sfizz.vst3/Contents/x86_64-linux/sfizz.so
#2  0x00007ffff6cc2fe4 in ags_vst3_plugin_load_plugin (base_plugin=0x5555557eb260) at ags/plugin/ags_vst3_plugin.c:591
```

**Comment by joelkraehemann** [comment]
> actually it's a SEGV
> 
> ```
> Thread 4 "gsequencer" received signal SIGSEGV, Segmentation fault.
> [Switching to Thread 0x7fffe77fe700 (LWP 14)]
> 0x00007fffcc20d5a5 in ?? () from /app/extensions/Plugins/vst3/sfizz.vst3/Contents/x86_64-linux/sfizz.so
> (gdb) where
> #0  0x00007fffcc20d5a5 in  () at /app/extensions/Plugins/vst3/sfizz.vst3/Contents/x86_64-linux/sfizz.so
> #1  0x00007fffcc20f077 in  () at /app/extensions/Plugins/vst3/sfizz.vst3/Contents/x86_64-linux/sfizz.so
> #2  0x00007ffff6cc2fe4 in ags_vst3_plugin_load_plugin (base_plugin=0x5555557eb260) at ags/plugin/ags_vst3_plugin.c:591
> ```

I only tested the example plugins from Steinberg. Might be we have some unicode issues, now.

File: modules/boost.json [hunk 9]
```diff
@@ -0,0 +1,16 @@
+{
```
Comment (hfiguiere) [inline modules/boost.json:1]: you don't need this

File: org.nongnu.gsequencer.gsequencer.json [hunk 10]
```diff
@@ -154,24 +158,79 @@
...
+                    "path": "patches/gsequencer.patch"
+                },
+                {
+                    "type": "script",
+                    "dest-filename": "autogen.sh",
+                    "commands": [
+                        "autoreconf -vif"
+                    ]
                 },
                 {
```
Comment (hfiguiere) [inline org.nongnu.gsequencer.gsequencer.json:235]: the change in the post-install was meant to make this `source` unecessary. (it was illogical)

**Review by hfiguiere — CHANGES_REQUESTED** [review]

File: modules/boost.json (hunk 9 above)
Comment (joelkraehemann) [reply modules/boost.json:1]: I patched Makefile.am

File: org.nongnu.gsequencer.gsequencer.json (hunk 10 above)
Comment (joelkraehemann) [reply org.nongnu.gsequencer.gsequencer.json:235]: configure failed

**Comment by hfiguiere** [comment]
if I uninstall `sfizz`, gsequencer finishes loading.

If I try to use Surge, I get this flooding the console:
```
** (gsequencer:5): WARNING **: 23:03:04.069: failed to create VST3 instance with plugin factory - IEditController
```
Surge is `org.freedesktop.LinuxAudio.Plugins.Surge`

**Comment by joelkraehemann** [comment]
> if I uninstall `sfizz`, gsequencer finishes loading.
> 

I am able to load sfizz, now. The problem was IComponent and IEditController are the same instance. So there is no need to createInstance() but rather call queryInterface().

> If I try to use Surge, I get this flooding the console:
> 
> ```
> ** (gsequencer:5): WARNING **: 23:03:04.069: failed to create VST3 instance with plugin factory - IEditController
> ```
> 
> Surge is `org.freedesktop.LinuxAudio.Plugins.Surge`

Looks like the same problem, but unsure for now.

File: org.nongnu.gsequencer.gsequencer.json (hunk 10 above)
Comment (joelkraehemann) [reply org.nongnu.gsequencer.gsequencer.json:223]: GPL-3 is showed within UI

File: modules/boost.json (hunk 9 above)
Comment (joelkraehemann) [reply modules/boost.json:1]: It was removed.

**Review by joelkraehemann — COMMENTED** [review]
thank you!

File: org.nongnu.gsequencer.gsequencer.json (hunk 10 above)
Comment (hfiguiere) [reply org.nongnu.gsequencer.gsequencer.json:223]: it is installed in the `post-install`.
This is for steps before the build.

**Comment by hfiguiere** [comment]
> > Surge is `org.freedesktop.LinuxAudio.Plugins.Surge`
> 
> Looks like the same problem, but unsure for now.

Still happening. With the latest.

**Pull request merged**